### ISSUE TITLE
`yaml-files` config option does not work

### ISSUE THREAD

**Issue by wojciech-galan** [adrienverge/yamllint#748]
I'd like to use `yamllint` for checking `cwl` files (they are subclass of yaml files). Unfortunatelly `yaml-files` in `.yamllint` doesn't seem to work and only *yml files are analyzed

Config file content:
```
---

yaml-files: 
  # - '*.yaml'
  # - '*.yml'
  # - '.yamllint'
  - '*.cwl'

rules:
  braces:
    level: warning
    max-spaces-inside: 1
  brackets:
    level: warning
    max-spaces-inside: 1
  colons:
    level: warning
  commas:
    level: warning
  comments: disable
  comments-indentation: disable
  document-start: disable
  empty-lines:
    level: warning
  hyphens:
    level: warning
  indentation:
    level: warning
    indent-sequences: consistent
  line-length:
    level: warning
    allow-non-breakable-inline-mappings: true
  truthy: disable
```

additional info:
```
$ python -V
Python 3.10.12
$ yamllint -v
yamllint 1.37.0
```

**Comment by andrewimeson** [comment]
I can't reproduce this.

1. Minimal config file (although I tested your whole file too)

   ```yaml
   ---
   # .yamllint
   yaml-files:
     - '*.yaml'
     - '*.yml'
     - '.yamllint'
     - '*.cwl'
   ```

2. Write some files with a lint issue

   ```bash
   for file in test.{yaml,yml,cwl}; do printf '---\nfile:  %s\n' "$file" > $file ; done
   ```

3. Run yamllint and get the errors in all the files, including the `test.cwl`

   ```console
   $ yamllint .
   ./test.cwl
     2:7       warning  too many spaces after colon  (colons)

   ./test.yaml
     2:7       warning  too many spaces after colon  (colons)

   ./test.yml
     2:7       warning  too many spaces after colon  (colons)
   ```

**Comment by wojciech-galan** [comment]
OK, now I know what was wrong with my execution. I misunderstood the docs and run it with `yamllint -d .yamllint .`

Thanks for help!
